spec/unit/provider/package/rpm_spec.rb

```diff
@@ -123,6 +123,13 @@
         provider.upgrade_package("ImageMagick-c++", "6.5.4.7-7.el6_5")
       end
 
+      it "should run rpm -U --oldpackage with the package source to downgrade" do
+        allow(new_resource).to receive(:allow_downgrade).and_return(true)
```

Comment:
you've got a real object here, its better to just `new_resource.allow_downgrade(true)` and not stub.